apis/v1alpha2/gateway_types.go

```diff
@@ -397,17 +397,18 @@ const (
 
 	// This reason is used with the "ResolvedRefs" condition when the
 	// Listener has a TLS configuration with at least one TLS CertificateRef
-	// that is invalid or cannot be resolved.
+	// that is malformed or does not exist.
```

Comment:
I'm not sure that "malformed" is clearer than "invalid" here. Trying to think through what could possibly cause this and it seems like it would either be:

- Not Found
- Nonexistent Kind
- Unsupported Kind

Anything else?

Reply:
It could also be used when the referenced secret data is invalid (it doesn't contain the data expected by the controller, or those data are malformed somehow).

Reply:
would you highlight these reasons specifically in the docs, instead of putting a generic `invalid` or `malformed` that isn't too meaningful?

Reply:
I think the ideal would be to add a sentence at the end of this paragraph that goes something like this: "For example, a CertificateRef is considered invalid when it refers to a nonexistent or unsupported resource or kind, or when the data within that resource is malformed."